cclient/serialization/json/json_serializer.c

```diff
@@ -247,8 +254,98 @@ retcode_t json_get_string(cJSON* json_obj, char* obj_name,
   return ret;
 }
 
+static retcode_t hash243_queue_to_json_array(hash243_queue_t queue,
+                                             cJSON* const json_root,
+                                             char const* const obj_name) {
+  size_t array_count;
+  cJSON* array_obj = NULL;
+
+  array_count = hash243_queue_count(&queue);
+  if (array_count > 0) {
+    array_obj = cJSON_CreateArray();
+    if (array_obj == NULL) {
+      return RC_CCLIENT_JSON_CREATE;
+    }
+    cJSON_AddItemToObject(json_root, obj_name, array_obj);
+
+    for (int i = 0; i < array_count; i++) {
```

Comment:
Can't we just loop over the queue instead of querying the ith element every time ?